fix: DIA-1927: [FE] Data on organization section are incorrect on new account settings page

nick-skriabin opened the pull request:
<!-- Follow Merge -->
This PR was automatically generated via Follow Merge. [Workflow Run](https://github.com/HumanSignal/label-studio/actions/runs/13760321176)
Please ensure that all linked upstream Pull Requests are merged before proceeding with this one.
- 
<!-- Follow Merge -->


Adds missing and fixes incorrect membership information on the User Account page.

Due to lack of contribution data in the API, this section is removed.

There's also difference between LSE and LSO in regard of what info is available via the API.

LSO
<img width="707" alt="image" src="https://github.com/user-attachments/assets/1feb2aeb-fccb-4273-a094-2e8863c128e8" />

LSE
<img width="703" alt="image" src="https://github.com/user-attachments/assets/16f568a3-463e-4087-b0bb-f0188291e90f" />

nick-skriabin committed:
Add missing membership info

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -1,45 +1,92 @@
+    if (!user?.date_joined) return null;
+    return format(new Date(user?.date_joined), "dd MMM yyyy, KK:mm a");
+  }, [user?.date_joined]);
+  console.log("====================");
```
```suggestion
```

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -1,45 +1,92 @@
+    queryKey: [user?.active_organization, user?.id, "user-membership"],
+    async queryFn() {
+      if (!user) return {};
+      console.log("loading membership");
```
```suggestion
```

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -1,45 +1,92 @@
-        }
+  const dateJoined = useMemo(() => {
+    if (!user?.date_joined) return null;
+    return format(new Date(user?.date_joined), "dd MMM yyyy, KK:mm a");
```
since we're formating the date in 2 spots we should create a reusable function so we can pass in the date that we want to format and get a formatted date back

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -1,45 +1,92 @@
+      });
+
+      console.log("user", response);
+
```
```suggestion
```

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -1,45 +1,92 @@
+    queryKey: ["organization", user?.active_organization],
+    async queryFn() {
+      if (!user) return null;
+      if (!window.APP_SETTINGS.billing) return null;
```
i've noticed that when doing tests this can cause issues - making it null safe avoids that
```suggestion
      if (!window?.APP_SETTINGS?.billing) return null;
```

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -1,45 +1,92 @@
       });
-  }, [user?.id]);
+      let role = "Owner";
+      console.log("org", organization);
```
```suggestion
```

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -1,45 +1,92 @@
+      console.log("org", organization);
+
+      switch (organization.userRole) {
+        case "OW":
+          role = "Owner";
+          break;
+        case "DI":
+          role = "Deactivated";
+          break;
+        case "AD":
+          role = "Administrator";
+          break;
+        case "MA":
+          role = "Manager";
+          break;
+        case "AN":
+          role = "Annotator";
+          break;
+        case "RE":
+          role = "Reviewer";
+          break;
+        case "NO":
+          role = "Pending";
+          break;
```
shouldn't this part be moved to a LSE override?

nick-skriabin replied:
this way I will have to override the entire section which is not ideal. having this mapping in LSO isn't too, though. apparently, must add a more granular module to override just a portion of the section

yyassi-heartex commented on web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx:
```diff
@@ -50,46 +97,52 @@ export const MembershipInfo = () => {
+      {/* <div className="flex gap-2 w-full justify-between"> */}
+      {/*   <div>Projects contributed to</div> */}
+      {/*   <div>{membership.data?.contributions}</div> */}
+      {/* </div> */}
```
if we dont need this lets remove it
```suggestion
```

nick-skriabin committed:
Update web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Update web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Update web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Update web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Reusable format function

nick-skriabin committed:
Update web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Update web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Update web/libs/core/src/pages/AccountSettings/sections/MembershipInfo.tsx

Co-authored-by: yyassi-heartex <104568407+yyassi-heartex@users.noreply.github.com>

nick-skriabin committed:
Update membership info

nick-skriabin committed:
Fix membership info

nick-skriabin committed:
Lift permissions for Org API

nick-skriabin committed:
Reset permission settings

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13648182690)
> Successfully merged:  140 files changed, 2456 insertions(+), 1026 deletions(-)

MihajloHoma committed:
Merge branch 'develop' into 'fb-dia-1927/membership-info'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13648182690

nick-skriabin committed:
Remove duplicated import

nick-skriabin committed:
Add `active_organization_meta` to whoami and support on user account page

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13697503711)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-1927/membership-info'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697503711

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697503711

nick-skriabin committed:
Safety first

nick-skriabin committed:
Merge branch 'develop' into fb-dia-1927/membership-info

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13715379207)

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13715379207

nick-skriabin committed:
Add missing import

nick-skriabin committed:
Restore version from develop

MihajloHoma commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13760321176)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-dia-1927/membership-info'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13760321176

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13760321176

nick-skriabin commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13764881655)
> Successfully merged:  5 files changed, 23 insertions(+), 55 deletions(-)

nick-skriabin committed:
Merge branch 'develop' into 'fb-dia-1927/membership-info'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13764881655
